backend/src/main/java/gov/cdc/usds/simplereport/api/model/filerow/ConditionAgnosticResultRow.java

```diff
@@ -0,0 +1,132 @@
+package gov.cdc.usds.simplereport.api.model.filerow;
+
+import static gov.cdc.usds.simplereport.validators.CsvValidatorUtils.*;
+
+import gov.cdc.usds.simplereport.config.FeatureFlagsConfig;
+import gov.cdc.usds.simplereport.service.ResultsUploaderCachingService;
+import gov.cdc.usds.simplereport.service.model.reportstream.FeedbackMessage;
+import gov.cdc.usds.simplereport.validators.CsvValidatorUtils.ValueOrError;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import lombok.Getter;
+
+@Getter
+public class ConditionAgnosticResultRow implements FileRow {
+  final ValueOrError testResultStatus;
+  final ValueOrError testResultEffectiveDate;
+  final ValueOrError testPerformedCode;
+  final ValueOrError testResultValue;
+  final ValueOrError patientId;
+  final ValueOrError patientFirstName;
+  final ValueOrError patientLastName;
+  final ValueOrError patientNameAbsentReason;
+  final ValueOrError patientAdminGender;
+
+  static final String TEST_RESULT_STATUS = "test_result_status";
+  static final String TEST_RESULT_EFFECTIVE_DATE = "test_result_effective_date";
+  static final String TEST_PERFORMED_CODE = "test_performed_code";
+  static final String TEST_RESULT_VALUE = "test_result_value";
+  static final String PATIENT_ID = "patient_id";
+  static final String PATIENT_LAST_NAME = "patient_last_name";
+  static final String PATIENT_FIRST_NAME = "patient_first_name";
+  static final String PATIENT_NAME_ABSENT_REASON = "patient_name_absent_reason";
+  static final String PATIENT_ADMIN_GENDER = "patient_admin_gender";
+
+  private static List<String> requiredFields =
+      List.of(
+          TEST_RESULT_STATUS,
+          TEST_RESULT_EFFECTIVE_DATE,
+          TEST_PERFORMED_CODE,
+          TEST_RESULT_VALUE,
+          PATIENT_ID,
+          PATIENT_ADMIN_GENDER);
+
+  public enum PatientNameRelatedField {
+    PATIENT_FIRST_NAME,
+    PATIENT_LAST_NAME,
+    PATIENT_NAME_ABSENT_REASON
+  }
+
+  @Override
+  public Boolean isRequired(String rowName) {
+    return requiredFields.contains(rowName);
+  }
+
+  //  overload for patient name related validations
+  private boolean isRequired(Map<String, String> rawRow, PatientNameRelatedField fieldToCheck) {
+    Map<String, Boolean> nameMap = new HashMap<>();
+    nameMap.put(PATIENT_FIRST_NAME, getValue(rawRow, PATIENT_FIRST_NAME, false).getValue() != null);
+    nameMap.put(PATIENT_LAST_NAME, getValue(rawRow, PATIENT_LAST_NAME, false).getValue() != null);
+    nameMap.put(
+        PATIENT_NAME_ABSENT_REASON,
+        getValue(rawRow, PATIENT_NAME_ABSENT_REASON, false).getValue() != null);
+
+    return switch (fieldToCheck) {
```

Comment:
Would appreciate extra attention to this block of code to make sure we're applying the restrictions correctly!